renderdoc/driver/gl/wrappers/gl_draw_funcs.cpp

```diff
@@ -269,6 +269,12 @@ void WrappedOpenGL::glDispatchCompute(GLuint num_groups_x, GLuint num_groups_y,
 {
   CoherentMapImplicitBarrier();
 
+  if(IsBackgroundCapturing(m_State))
```

Comment:
The background capturing case isn't the one we need to order before the GL call. In fact it's fine if during background capturing we update the references after making the call. The important one is during active capturing since we need to detect the write reference before the GL command actually performs the write, in order to do a last minute prepare of a postponed initial state.

Reply:
Hm, that's right.. Do we need just put the actual GL call then after any extra handling? For consistency below Background/Active capturing?

Reply:
As with the other functions we're serialising, we still need to do the serialisation _after_ the GL call so that the results of `SERIALISE_TIME_CALL` can be serialised with the timestamp & duration. The only thing I think that needs to happen before it is the `GLRenderState` work to mark referenced (which will then potentially trigger a last-second prepare). If you want to pull out the `GLRenderState` work for background capturing as well to keep the code tidy that's fine.

Reply:
OK, sounds good, will work on it. 

Just to think on the same page, is this is what you have in mind? :

```cpp
void WrappedOpenGL::glDispatchComputeIndirect(GLintptr indirect)
{
  CoherentMapImplicitBarrier();

  if(IsBackgroundCapturing(m_State))
  {
    GLRenderState state;
    state.MarkDirty(this);
  }
  else if(IsActiveCapturing(m_State))
  {
    GLRenderState state;
    state.FetchState(this);
    state.MarkReferenced(this, false);
  }

  SERIALISE_TIME_CALL(GL.glDispatchComputeIndirect(indirect));

  if(IsActiveCapturing(m_State))
  {
    USE_SCRATCH_SERIALISER();
    ser.SetDrawChunk();
    SCOPED_SERIALISE_CHUNK(gl_CurChunk);
    Serialise_glDispatchComputeIndirect(ser, indirect);

    GetContextRecord()->AddChunk(scope.Get());
  }
}
```

Reply:
Yep I think that looks good.